teamcity: failed tests on master: testrace/TestScanBatches, test/TestLogic, test/TestLogic: TestLogic/distsql, test/TestLogic: TestLogic/distsql: TestLogic/distsql/select_index, test/TestLogic, test/TestLogic: TestLogic/distsql, test/TestLogic: TestLogic/distsql: TestLogic/distsql/select_index, testrace/TestScanBatches

The following tests appear to have failed:

[#523139](https://teamcity.cockroachdb.com/viewLog.html?buildId=523139):


```
--- FAIL: testrace/TestScanBatches (0.000s)
Race detected!

------- Stdout: -------
W180215 20:45:47.379664 25430 server/status/runtime.go:109  Could not parse build timestamp: parsing time "" as "2006/01/02 15:04:05": cannot parse "" as "2006"
I180215 20:45:47.656895 25430 server/config.go:522  [n?] 1 storage engine initialized
I180215 20:45:47.656995 25430 server/config.go:525  [n?] RocksDB cache size: 128 MiB
I180215 20:45:47.657049 25430 server/config.go:525  [n?] store 0: in-memory, size 0 B
I180215 20:45:47.746570 25430 server/node.go:374  [n?] **** cluster 73107500-011e-490e-b0c6-053a9b2d599b has been created
I180215 20:45:47.746799 25430 server/server.go:979  [n?] **** add additional nodes by specifying --join=127.0.0.1:33565
I180215 20:45:47.799317 25430 storage/store.go:1307  [n1,s1] [n1,s1]: failed initial metrics computation: [n1,s1]: system config not yet available
I180215 20:45:47.835731 25430 server/node.go:504  [n1] initialized store [n1,s1]: disk (capacity=512 MiB, available=512 MiB, used=0 B, logicalBytes=4.9 KiB), ranges=1, leases=1, writes=0.00, bytesPerReplica={p10=5045.00 p25=5045.00 p50=5045.00 p75=5045.00 p90=5045.00}, writesPerReplica={p10=0.00 p25=0.00 p50=0.00 p75=0.00 p90=0.00}
I180215 20:45:47.836277 25430 server/node.go:352  [n1] node ID 1 initialized
I180215 20:45:47.836858 25430 gossip/gossip.go:332  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:33565" > attrs:<> locality:<> ServerVersion:<major_val:1 minor_val:1 patch:0 unstable:12 > 
I180215 20:45:47.838379 25430 storage/stores.go:331  [n1] read 0 node addresses from persistent storage
I180215 20:45:47.839088 25430 server/node.go:645  [n1] connecting to gossip network to verify cluster ID...
I180215 20:45:47.839311 25430 server/node.go:670  [n1] node connected via gossip and verified as part of cluster "73107500-011e-490e-b0c6-053a9b2d599b"
I180215 20:45:47.839876 25430 server/node.go:446  [n1] node=1: started with [<no-attributes>=<in-mem>] engine(s) and attributes []
I180215 20:45:47.843224 25430 server/server.go:1180  [n1] starting https server at 127.0.0.1:39015
I180215 20:45:47.843453 25430 server/server.go:1181  [n1] starting grpc/postgres server at 127.0.0.1:33565
I180215 20:45:47.879364 25430 server/server.go:1182  [n1] advertising CockroachDB node at 127.0.0.1:33565
W180215 20:45:47.881231 25430 sql/jobs/registry.go:233  [n1] unable to get node liveness: node not in the liveness table
I180215 20:45:47.941101 25666 storage/replica_command.go:838  [split,n1,s1,r1/1:/M{in-ax}] initiating a split of this range at key /System/"" [r2]
I180215 20:45:48.135281 25370 storage/replica_command.go:838  [split,n1,s1,r2/1:/{System/-Max}] initiating a split of this range at key /System/NodeLiveness [r3]
W180215 20:45:48.159209 25405 storage/intent_resolver.go:320  [n1,s1] failed to push during intent resolution: failed to push "sql txn implicit" id=eb00cfa8 key=/Table/SystemConfigSpan/Start rw=true pri=0.01152488 iso=SERIALIZABLE stat=PENDING epo=0 ts=1518727547.977372801,0 orig=1518727547.977372801,0 max=1518727547.977372801,0 wto=false rop=false seq=7
I180215 20:45:48.261986 25430 sql/event_log.go:114  [n1] Event: "set_cluster_setting", target: 0, info: {SettingName:diagnostics.reporting.enabled Value:true User:node}
I180215 20:45:48.670132 25421 storage/replica_command.go:838  [split,n1,s1,r3/1:/{System/NodeL…-Max}] initiating a split of this range at key /System/NodeLivenessMax [r4]
I180215 20:45:48.716440 25420 storage/replica_consistency.go:129  [replica consistency checker,n1,s1,r2/1:/System/{-NodeLive…}] triggering stats recomputation to resolve delta of {ContainsEstimates:true LastUpdateNanos:1518727548388465725 IntentAge:0 GCBytesAge:0 LiveBytes:-21744 LiveCount:-430 KeyBytes:-20024 KeyCount:-430 ValBytes:-1720 ValCount:-430 IntentBytes:0 IntentCount:0 SysBytes:0 SysCount:0}
I180215 20:45:48.888639 25371 storage/replica_command.go:838  [split,n1,s1,r4/1:/{System/NodeL…-Max}] initiating a split of this range at key /System/tsd [r5]
W180215 20:45:48.898731 25655 storage/intent_resolver.go:320  [n1,s1] failed to push during intent resolution: failed to push "sql txn implicit" id=26abfda2 key=/Table/SystemConfigSpan/Start rw=true pri=0.02592397 iso=SERIALIZABLE stat=PENDING epo=0 ts=1518727548.600813294,0 orig=1518727548.600813294,0 max=1518727548.600813294,0 wto=false rop=false seq=16
I180215 20:45:49.037774 25430 sql/event_log.go:114  [n1] Event: "set_cluster_setting", target: 0, info: {SettingName:version Value:$1 User:node}
I180215 20:45:49.118279 25719 storage/replica_command.go:838  [split,n1,s1,r5/1:/{System/tsd-Max}] initiating a split of this range at key /System/"tse" [r6]
W180215 20:45:49.153087 25553 storage/intent_resolver.go:320  [n1,s1] failed to push during intent resolution: failed to push "sql txn implicit" id=79ee019d key=/Table/SystemConfigSpan/Start rw=true pri=0.00158185 iso=SERIALIZABLE stat=PENDING epo=0 ts=1518727549.045174703,0 orig=1518727549.045174703,0 max=1518727549.045174703,0 wto=false rop=false seq=7
I180215 20:45:49.264385 25430 sql/event_log.go:114  [n1] Event: "set_cluster_setting", target: 0, info: {SettingName:trace.debug.enable Value:false User:node}
I180215 20:45:49.424403 25374 storage/replica_command.go:838  [split,n1,s1,r6/1:/{System/tse-Max}] initiating a split of this range at key /Table/SystemConfigSpan/Start [r7]
I180215 20:45:49.566944 25779 storage/replica_command.go:838  [split,n1,s1,r7/1:/{Table/System…-Max}] initiating a split of this range at key /Table/11 [r8]
I180215 20:45:49.718149 25740 storage/replica_command.go:838  [split,n1,s1,r8/1:/{Table/11-Max}] initiating a split of this range at key /Table/12 [r9]
W180215 20:45:49.864930 25773 storage/intent_resolver.go:320  [n1,s1] failed to push during intent resolution: failed to push "split" id=4437c7c2 key=/Local/Range/Table/11/RangeDescriptor rw=true pri=0.04722760 iso=SERIALIZABLE stat=PENDING epo=0 ts=1518727549.718352692,0 orig=1518727549.718352692,0 max=1518727549.718352692,0 wto=false rop=false seq=3
I180215 20:45:49.880724 25724 storage/replica_command.go:838  [split,n1,s1,r9/1:/{Table/12-Max}] initiating a split of this range at key /Table/13 [r10]
I180215 20:45:50.064593 25799 storage/replica_command.go:838  [split,n1,s1,r10/1:/{Table/13-Max}] initiating a split of this range at key /Table/14 [r11]
I180215 20:45:50.233259 25430 sql/event_log.go:114  [n1] Event: "alter_table", target: 4, info: {TableName:users Statement:ALTER TABLE system.users ADD COLUMN IF NOT EXISTS "isRole" BOOL NOT NULL DEFAULT false User:node MutationID:1 CascadeDroppedViews:[]}
I180215 20:45:50.288276 25783 storage/replica_command.go:838  [split,n1,s1,r11/1:/{Table/14-Max}] initiating a split of this range at key /Table/15 [r12]
I180215 20:45:50.315977 25430 sql/lease.go:349  [n1] publish: descID=4 (users) version=2 mtime=2018-02-15 20:45:50.314602461 +0000 UTC
I180215 20:45:50.450349 25831 storage/replica_command.go:838  [split,n1,s1,r12/1:/{Table/15-Max}] initiating a split of this range at key /Table/16 [r13]
I180215 20:45:50.591614 25430 sql/lease.go:349  [n1] publish: descID=4 (users) version=3 mtime=2018-02-15 20:45:50.572612115 +0000 UTC
W180215 20:45:50.646122 25803 storage/intent_resolver.go:320  [n1,s1] failed to push during intent resolution: failed to push "split" id=7e6fc168 key=/Local/Range/Table/15/RangeDescriptor rw=true pri=0.00128914 iso=SERIALIZABLE stat=PENDING epo=0 ts=1518727550.450587224,0 orig=1518727550.450587224,0 max=1518727550.450587224,0 wto=false rop=false seq=3
I180215 20:45:50.702722 25804 storage/replica_command.go:838  [split,n1,s1,r13/1:/{Table/16-Max}] initiating a split of this range at key /Table/17 [r14]
I180215 20:45:50.880666 25430 sql/backfill.go:134  [n1] Running backfill for "users", v=3, m=1
I180215 20:45:50.893823 25855 storage/replica_command.go:838  [split,n1,s1,r14/1:/{Table/17-Max}] initiating a split of this range at key /Table/18 [r15]
W180215 20:45:50.975516 25809 storage/intent_resolver.go:320  [n1,s1] failed to push during intent resolution: failed to push "split" id=a9a025d9 key=/Local/Range/Table/17/RangeDescriptor rw=true pri=0.00999388 iso=SERIALIZABLE stat=PENDING epo=0 ts=1518727550.894114701,0 orig=1518727550.894114701,0 max=1518727550.894114701,0 wto=false rop=false seq=3
I180215 20:45:50.992620 25838 storage/replica_command.go:838  [split,n1,s1,r15/1:/{Table/18-Max}] initiating a split of this range at key /Table/19 [r16]
I180215 20:45:51.142605 25777 storage/replica_command.go:838  [split,n1,s1,r16/1:/{Table/19-Max}] initiating a split of this range at key /Table/20 [r17]
I180215 20:45:51.273715 25943 storage/replica_command.go:838  [split,n1,s1,r17/1:/{Table/20-Max}] initiating a split of this range at key /Table/21 [r18]
I180215 20:45:51.468898 25790 storage/replica_command.go:838  [split,n1,s1,r18/1:/{Table/21-Max}] initiating a split of this range at key /Table/22 [r19]
I180215 20:45:51.504783 25430 sql/lease.go:349  [n1] publish: descID=4 (users) version=4 mtime=2018-02-15 20:45:51.489653481 +0000 UTC
I180215 20:45:51.630626 25912 storage/replica_command.go:838  [split,n1,s1,r19/1:/{Table/22-Max}] initiating a split of this range at key /Table/23 [r20]
W180215 20:45:51.786895 25916 storage/intent_resolver.go:320  [n1,s1] failed to push during intent resolution: failed to push "split" id=164ad23f key=/Local/Range/Table/22/RangeDescriptor rw=true pri=0.00522989 iso=SERIALIZABLE stat=PENDING epo=0 ts=1518727551.630845456,0 orig=1518727551.630845456,0 max=1518727551.630845456,0 wto=false rop=false seq=3
I180215 20:45:51.822392 25430 sql/event_log.go:114  [n1] Event: "finish_schema_change", target: 4, info: {MutationID:1}
I180215 20:45:51.856140 25430 sql/lease.go:275  publish (count leases): descID=4 name=users version=3 count=1
I180215 20:45:51.887307 25989 sql/lease.go:800  new lease: 4("users") ver=4:1518727806.434029774,0, refcount=0
I180215 20:45:51.931253 25430 sql/lease.go:275  publish (count leases): descID=4 name=users version=3 count=1
I180215 20:45:52.523151 25430 server/server.go:1254  [n1] done ensuring all necessary migrations have run
I180215 20:45:52.523346 25430 server/server.go:1257  [n1] serving sql connections
I180215 20:45:52.628929 25952 sql/event_log.go:114  [n1] Event: "node_join", target: 1, info: {Descriptor:{NodeID:1 Address:{NetworkField:tcp AddressField:127.0.0.1:33565} Attrs: Locality: ServerVersion:1.1-12} ClusterID:73107500-011e-490e-b0c6-053a9b2d599b StartedAt:1518727547839393362 LastUp:1518727547839393362}
I180215 20:45:52.993902 25964 storage/replica_command.go:838  [split,n1,s1,r20/1:/{Table/23-Max}] initiating a split of this range at key /Table/50 [r21]
I180215 20:45:53.029558 25962 sql/event_log.go:114  [client=127.0.0.1:60458,user=root,n1] Event: "create_database", target: 50, info: {DatabaseName:test Statement:CREATE DATABASE IF NOT EXISTS test User:root}
I180215 20:45:53.224882 25965 storage/replica_command.go:838  [split,n1,s1,r21/1:/{Table/50-Max}] initiating a split of this range at key /Table/51 [r22]
I180215 20:45:53.243403 25962 sql/event_log.go:114  [client=127.0.0.1:60458,user=root,n1] Event: "create_table", target: 51, info: {TableName:test.scan Statement:CREATE TABLE test.scan (a INT, b INT, v STRING, PRIMARY KEY (a, b)) User:root}
W180215 20:45:53.380561 26050 storage/intent_resolver.go:320  [n1,s1] failed to push during intent resolution: failed to push "split" id=1b04c15a key=/Local/Range/Table/50/RangeDescriptor rw=true pri=0.01770058 iso=SERIALIZABLE stat=PENDING epo=0 ts=1518727553.225189310,0 orig=1518727553.225189310,0 max=1518727553.225189310,0 wto=false rop=false seq=3
I180215 20:45:56.308420 25590 storage/replica_proposal.go:196  [n1,s1,r22/1:/{Table/51-Max}] new range lease repl=(n1,s1):1 seq=2 start=0.000000000,0 epo=1 pro=1518727556.304102206,0 following repl=(n1,s1):1 seq=1 start=0.000000000,0 exp=1518727556.800523088,0 pro=1518727547.800616679,0
I180215 20:45:56.397743 25572 storage/replica_proposal.go:196  [n1,s1,r12/1:/Table/1{5-6}] new range lease repl=(n1,s1):1 seq=2 start=0.000000000,0 epo=1 pro=1518727556.393261485,0 following repl=(n1,s1):1 seq=1 start=0.000000000,0 exp=1518727556.800523088,0 pro=1518727547.800616679,0
I180215 20:45:56.599722 25585 storage/replica_proposal.go:196  [n1,s1,r18/1:/Table/2{1-2}] new range lease repl=(n1,s1):1 seq=2 start=0.000000000,0 epo=1 pro=1518727556.595558958,0 following repl=(n1,s1):1 seq=1 start=0.000000000,0 exp=1518727556.800523088,0 pro=1518727547.800616679,0
I180215 20:45:56.810332 25604 storage/replica_proposal.go:196  [n1,s1,r17/1:/Table/2{0-1}] new range lease repl=(n1,s1):1 seq=2 start=0.000000000,0 epo=1 pro=1518727556.802097983,0 following repl=(n1,s1):1 seq=1 start=0.000000000,0 exp=1518727556.800523088,0 pro=1518727547.800616679,0
I180215 20:45:57.065767 25593 storage/replica_proposal.go:196  [n1,s1,r7/1:/Table/{SystemCon…-11}] new range lease repl=(n1,s1):1 seq=2 start=0.000000000,0 epo=1 pro=1518727557.061344807,0 following repl=(n1,s1):1 seq=1 start=0.000000000,0 exp=1518727556.800523088,0 pro=1518727547.800616679,0
==================
Write at 0x000003dee838 by goroutine 862:
  github.com/cockroachdb/cockroach/pkg/sql/sqlbase.SetKVBatchSize()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/sqlbase/kvfetcher.go:138 +0x5d
  github.com/cockroachdb/cockroach/pkg/sql.TestScanBatches()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/scan_test.go:178 +0xbeb
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:746 +0x16c

Previous read at 0x000003dee838 by goroutine 550:
  github.com/cockroachdb/cockroach/pkg/sql/sqlbase.(*txnKVFetcher).fetch()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/sqlbase/kvfetcher.go:268 +0x1324
  github.com/cockroachdb/cockroach/pkg/sql/sqlbase.(*txnKVFetcher).nextBatch()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/sqlbase/kvfetcher.go:359 +0xbb
  github.com/cockroachdb/cockroach/pkg/sql/sqlbase.(*RowFetcher).nextKV()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/sqlbase/rowfetcher.go:419 +0x22e
  github.com/cockroachdb/cockroach/pkg/sql/sqlbase.(*RowFetcher).NextKey()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/sqlbase/rowfetcher.go:435 +0x98
  github.com/cockroachdb/cockroach/pkg/sql/sqlbase.(*RowFetcher).StartScanFrom()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/sqlbase/rowfetcher.go:409 +0x111
  github.com/cockroachdb/cockroach/pkg/sql/sqlbase.(*RowFetcher).StartScan()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/sqlbase/rowfetcher.go:399 +0x29a
  github.com/cockroachdb/cockroach/pkg/sql.(*scanNode).initScan()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/scan.go:222 +0x203
  github.com/cockroachdb/cockroach/pkg/sql.(*scanNode).Next()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/scan.go:182 +0x486
  github.com/cockroachdb/cockroach/pkg/sql.(*renderNode).Next()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/render.go:338 +0xad
  github.com/cockroachdb/cockroach/pkg/sql.(*groupNode).Next()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/group.go:362 +0xead
  github.com/cockroachdb/cockroach/pkg/sql.forEachRow()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/executor.go:1978 +0x79
  github.com/cockroachdb/cockroach/pkg/sql.(*planner).queryRows()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/planner.go:380 +0x497
  github.com/cockroachdb/cockroach/pkg/sql.(*planner).QueryRow()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/planner.go:338 +0xab
  github.com/cockroachdb/cockroach/pkg/sql.LeaseStore.countLeases.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/lease.go:405 +0x36c
  github.com/cockroachdb/cockroach/pkg/internal/client.(*DB).Txn.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/db.go:515 +0x5e
  github.com/cockroachdb/cockroach/pkg/internal/client.(*Txn).Exec()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/txn.go:738 +0x128
  github.com/cockroachdb/cockroach/pkg/internal/client.(*DB).Txn()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/db.go:514 +0x183
  github.com/cockroachdb/cockroach/pkg/sql.LeaseStore.countLeases()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/lease.go:399 +0x1f0
  github.com/cockroachdb/cockroach/pkg/sql.LeaseStore.WaitForOneVersion()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/lease.go:268 +0x74c
  github.com/cockroachdb/cockroach/pkg/sql.LeaseStore.Publish()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/lease.go:304 +0x224
  github.com/cockroachdb/cockroach/pkg/sql.(*SchemaChanger).MaybeIncrementVersion()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/schema_changer.go:509 +0x122
  github.com/cockroachdb/cockroach/pkg/sql.(*SchemaChanger).exec()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/schema_changer.go:396 +0x1e4
  github.com/cockroachdb/cockroach/pkg/sql.(*SchemaChangeManager).Start.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/schema_changer.go:1084 +0x9d6
  github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:195 +0x160

Goroutine 862 (running) created at:
  testing.(*T).Run()
      /usr/local/go/src/testing/testing.go:789 +0x568
  testing.runTests.func1()
      /usr/local/go/src/testing/testing.go:1004 +0xa7
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:746 +0x16c
  testing.runTests()
      /usr/local/go/src/testing/testing.go:1002 +0x521
  testing.(*M).Run()
      /usr/local/go/src/testing/testing.go:921 +0x206
  github.com/cockroachdb/cockroach/pkg/sql_test.TestMain()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/main_test.go:36 +0x15d
  main.main()
      github.com/cockroachdb/cockroach/pkg/sql/_test/_testmain.go:480 +0x1cd

Goroutine 550 (running) created at:
  github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:188 +0xba
  github.com/cockroachdb/cockroach/pkg/sql.(*SchemaChangeManager).Start()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/schema_changer.go:979 +0x12f
  github.com/cockroachdb/cockroach/pkg/server.(*Server).Start()
      /go/src/github.com/cockroachdb/cockroach/pkg/server/server.go:1165 +0x43e0
  github.com/cockroachdb/cockroach/pkg/server.(*TestServer).Start()
      /go/src/github.com/cockroachdb/cockroach/pkg/server/testserver.go:336 +0x24c
  github.com/cockroachdb/cockroach/pkg/testutils/serverutils.StartServerRaw()
      /go/src/github.com/cockroachdb/cockroach/pkg/testutils/serverutils/test_server_shim.go:204 +0x143
  github.com/cockroachdb/cockroach/pkg/testutils/serverutils.StartServer()
      /go/src/github.com/cockroachdb/cockroach/pkg/testutils/serverutils/test_server_shim.go:172 +0x73
  github.com/cockroachdb/cockroach/pkg/sql.TestScanBatches()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/scan_test.go:131 +0xc5
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:746 +0x16c
==================
?   	github.com/cockroachdb/cockroach/pkg/sql/coltypes	[no test files]
I180215 20:40:52.815320 1 rand.go:74  Random seed: 3295516578724522379
--- FAIL: test/TestLogic (73.580s)
test_log_scope.go:81: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestLogic376239698
test_log_scope.go:62: use -show-logs to present logs inline
--- FAIL: test/TestLogic: TestLogic/distsql (0.320s)
None
--- FAIL: test/TestLogic: TestLogic/distsql: TestLogic/distsql/select_index (0.920s)
logic.go:2095: 
	testdata/logic_test/select_index:1033: SELECT * FROM xy WHERE x IN (NULL, 1, 2)
	expected:
	    1  NULL
	    1  1
	    
	but found (query options: "") :
	    1  1     
	    1  NULL  
	
logic.go:2126: testdata/logic_test/select_index:1039: too many errors encountered, skipping the rest of the input
--- FAIL: test/TestLogic (73.580s)
test_log_scope.go:81: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestLogic376239698
test_log_scope.go:62: use -show-logs to present logs inline
--- FAIL: test/TestLogic: TestLogic/distsql (0.320s)
None
--- FAIL: test/TestLogic: TestLogic/distsql: TestLogic/distsql/select_index (0.920s)
logic.go:2095: 
	testdata/logic_test/select_index:1033: SELECT * FROM xy WHERE x IN (NULL, 1, 2)
	expected:
	    1  NULL
	    1  1
	    
	but found (query options: "") :
	    1  1     
	    1  NULL  
	
logic.go:2126: testdata/logic_test/select_index:1039: too many errors encountered, skipping the rest of the input
--- FAIL: testrace/TestScanBatches (0.440s)
Race detected!

------- Stdout: -------
W180215 20:45:47.379664 25430 server/status/runtime.go:109  Could not parse build timestamp: parsing time "" as "2006/01/02 15:04:05": cannot parse "" as "2006"
I180215 20:45:47.656895 25430 server/config.go:522  [n?] 1 storage engine initialized
I180215 20:45:47.656995 25430 server/config.go:525  [n?] RocksDB cache size: 128 MiB
I180215 20:45:47.657049 25430 server/config.go:525  [n?] store 0: in-memory, size 0 B
I180215 20:45:47.746570 25430 server/node.go:374  [n?] **** cluster 73107500-011e-490e-b0c6-053a9b2d599b has been created
I180215 20:45:47.746799 25430 server/server.go:979  [n?] **** add additional nodes by specifying --join=127.0.0.1:33565
I180215 20:45:47.799317 25430 storage/store.go:1307  [n1,s1] [n1,s1]: failed initial metrics computation: [n1,s1]: system config not yet available
I180215 20:45:47.835731 25430 server/node.go:504  [n1] initialized store [n1,s1]: disk (capacity=512 MiB, available=512 MiB, used=0 B, logicalBytes=4.9 KiB), ranges=1, leases=1, writes=0.00, bytesPerReplica={p10=5045.00 p25=5045.00 p50=5045.00 p75=5045.00 p90=5045.00}, writesPerReplica={p10=0.00 p25=0.00 p50=0.00 p75=0.00 p90=0.00}
I180215 20:45:47.836277 25430 server/node.go:352  [n1] node ID 1 initialized
I180215 20:45:47.836858 25430 gossip/gossip.go:332  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:33565" > attrs:<> locality:<> ServerVersion:<major_val:1 minor_val:1 patch:0 unstable:12 > 
I180215 20:45:47.838379 25430 storage/stores.go:331  [n1] read 0 node addresses from persistent storage
I180215 20:45:47.839088 25430 server/node.go:645  [n1] connecting to gossip network to verify cluster ID...
I180215 20:45:47.839311 25430 server/node.go:670  [n1] node connected via gossip and verified as part of cluster "73107500-011e-490e-b0c6-053a9b2d599b"
I180215 20:45:47.839876 25430 server/node.go:446  [n1] node=1: started with [<no-attributes>=<in-mem>] engine(s) and attributes []
I180215 20:45:47.843224 25430 server/server.go:1180  [n1] starting https server at 127.0.0.1:39015
I180215 20:45:47.843453 25430 server/server.go:1181  [n1] starting grpc/postgres server at 127.0.0.1:33565
I180215 20:45:47.879364 25430 server/server.go:1182  [n1] advertising CockroachDB node at 127.0.0.1:33565
W180215 20:45:47.881231 25430 sql/jobs/registry.go:233  [n1] unable to get node liveness: node not in the liveness table
I180215 20:45:47.941101 25666 storage/replica_command.go:838  [split,n1,s1,r1/1:/M{in-ax}] initiating a split of this range at key /System/"" [r2]
I180215 20:45:48.135281 25370 storage/replica_command.go:838  [split,n1,s1,r2/1:/{System/-Max}] initiating a split of this range at key /System/NodeLiveness [r3]
W180215 20:45:48.159209 25405 storage/intent_resolver.go:320  [n1,s1] failed to push during intent resolution: failed to push "sql txn implicit" id=eb00cfa8 key=/Table/SystemConfigSpan/Start rw=true pri=0.01152488 iso=SERIALIZABLE stat=PENDING epo=0 ts=1518727547.977372801,0 orig=1518727547.977372801,0 max=1518727547.977372801,0 wto=false rop=false seq=7
I180215 20:45:48.261986 25430 sql/event_log.go:114  [n1] Event: "set_cluster_setting", target: 0, info: {SettingName:diagnostics.reporting.enabled Value:true User:node}
I180215 20:45:48.670132 25421 storage/replica_command.go:838  [split,n1,s1,r3/1:/{System/NodeL…-Max}] initiating a split of this range at key /System/NodeLivenessMax [r4]
I180215 20:45:48.716440 25420 storage/replica_consistency.go:129  [replica consistency checker,n1,s1,r2/1:/System/{-NodeLive…}] triggering stats recomputation to resolve delta of {ContainsEstimates:true LastUpdateNanos:1518727548388465725 IntentAge:0 GCBytesAge:0 LiveBytes:-21744 LiveCount:-430 KeyBytes:-20024 KeyCount:-430 ValBytes:-1720 ValCount:-430 IntentBytes:0 IntentCount:0 SysBytes:0 SysCount:0}
I180215 20:45:48.888639 25371 storage/replica_command.go:838  [split,n1,s1,r4/1:/{System/NodeL…-Max}] initiating a split of this range at key /System/tsd [r5]
W180215 20:45:48.898731 25655 storage/intent_resolver.go:320  [n1,s1] failed to push during intent resolution: failed to push "sql txn implicit" id=26abfda2 key=/Table/SystemConfigSpan/Start rw=true pri=0.02592397 iso=SERIALIZABLE stat=PENDING epo=0 ts=1518727548.600813294,0 orig=1518727548.600813294,0 max=1518727548.600813294,0 wto=false rop=false seq=16
I180215 20:45:49.037774 25430 sql/event_log.go:114  [n1] Event: "set_cluster_setting", target: 0, info: {SettingName:version Value:$1 User:node}
I180215 20:45:49.118279 25719 storage/replica_command.go:838  [split,n1,s1,r5/1:/{System/tsd-Max}] initiating a split of this range at key /System/"tse" [r6]
W180215 20:45:49.153087 25553 storage/intent_resolver.go:320  [n1,s1] failed to push during intent resolution: failed to push "sql txn implicit" id=79ee019d key=/Table/SystemConfigSpan/Start rw=true pri=0.00158185 iso=SERIALIZABLE stat=PENDING epo=0 ts=1518727549.045174703,0 orig=1518727549.045174703,0 max=1518727549.045174703,0 wto=false rop=false seq=7
I180215 20:45:49.264385 25430 sql/event_log.go:114  [n1] Event: "set_cluster_setting", target: 0, info: {SettingName:trace.debug.enable Value:false User:node}
I180215 20:45:49.424403 25374 storage/replica_command.go:838  [split,n1,s1,r6/1:/{System/tse-Max}] initiating a split of this range at key /Table/SystemConfigSpan/Start [r7]
I180215 20:45:49.566944 25779 storage/replica_command.go:838  [split,n1,s1,r7/1:/{Table/System…-Max}] initiating a split of this range at key /Table/11 [r8]
I180215 20:45:49.718149 25740 storage/replica_command.go:838  [split,n1,s1,r8/1:/{Table/11-Max}] initiating a split of this range at key /Table/12 [r9]
W180215 20:45:49.864930 25773 storage/intent_resolver.go:320  [n1,s1] failed to push during intent resolution: failed to push "split" id=4437c7c2 key=/Local/Range/Table/11/RangeDescriptor rw=true pri=0.04722760 iso=SERIALIZABLE stat=PENDING epo=0 ts=1518727549.718352692,0 orig=1518727549.718352692,0 max=1518727549.718352692,0 wto=false rop=false seq=3
I180215 20:45:49.880724 25724 storage/replica_command.go:838  [split,n1,s1,r9/1:/{Table/12-Max}] initiating a split of this range at key /Table/13 [r10]
I180215 20:45:50.064593 25799 storage/replica_command.go:838  [split,n1,s1,r10/1:/{Table/13-Max}] initiating a split of this range at key /Table/14 [r11]
I180215 20:45:50.233259 25430 sql/event_log.go:114  [n1] Event: "alter_table", target: 4, info: {TableName:users Statement:ALTER TABLE system.users ADD COLUMN IF NOT EXISTS "isRole" BOOL NOT NULL DEFAULT false User:node MutationID:1 CascadeDroppedViews:[]}
I180215 20:45:50.288276 25783 storage/replica_command.go:838  [split,n1,s1,r11/1:/{Table/14-Max}] initiating a split of this range at key /Table/15 [r12]
I180215 20:45:50.315977 25430 sql/lease.go:349  [n1] publish: descID=4 (users) version=2 mtime=2018-02-15 20:45:50.314602461 +0000 UTC
I180215 20:45:50.450349 25831 storage/replica_command.go:838  [split,n1,s1,r12/1:/{Table/15-Max}] initiating a split of this range at key /Table/16 [r13]
I180215 20:45:50.591614 25430 sql/lease.go:349  [n1] publish: descID=4 (users) version=3 mtime=2018-02-15 20:45:50.572612115 +0000 UTC
W180215 20:45:50.646122 25803 storage/intent_resolver.go:320  [n1,s1] failed to push during intent resolution: failed to push "split" id=7e6fc168 key=/Local/Range/Table/15/RangeDescriptor rw=true pri=0.00128914 iso=SERIALIZABLE stat=PENDING epo=0 ts=1518727550.450587224,0 orig=1518727550.450587224,0 max=1518727550.450587224,0 wto=false rop=false seq=3
I180215 20:45:50.702722 25804 storage/replica_command.go:838  [split,n1,s1,r13/1:/{Table/16-Max}] initiating a split of this range at key /Table/17 [r14]
I180215 20:45:50.880666 25430 sql/backfill.go:134  [n1] Running backfill for "users", v=3, m=1
I180215 20:45:50.893823 25855 storage/replica_command.go:838  [split,n1,s1,r14/1:/{Table/17-Max}] initiating a split of this range at key /Table/18 [r15]
W180215 20:45:50.975516 25809 storage/intent_resolver.go:320  [n1,s1] failed to push during intent resolution: failed to push "split" id=a9a025d9 key=/Local/Range/Table/17/RangeDescriptor rw=true pri=0.00999388 iso=SERIALIZABLE stat=PENDING epo=0 ts=1518727550.894114701,0 orig=1518727550.894114701,0 max=1518727550.894114701,0 wto=false rop=false seq=3
I180215 20:45:50.992620 25838 storage/replica_command.go:838  [split,n1,s1,r15/1:/{Table/18-Max}] initiating a split of this range at key /Table/19 [r16]
I180215 20:45:51.142605 25777 storage/replica_command.go:838  [split,n1,s1,r16/1:/{Table/19-Max}] initiating a split of this range at key /Table/20 [r17]
I180215 20:45:51.273715 25943 storage/replica_command.go:838  [split,n1,s1,r17/1:/{Table/20-Max}] initiating a split of this range at key /Table/21 [r18]
I180215 20:45:51.468898 25790 storage/replica_command.go:838  [split,n1,s1,r18/1:/{Table/21-Max}] initiating a split of this range at key /Table/22 [r19]
I180215 20:45:51.504783 25430 sql/lease.go:349  [n1] publish: descID=4 (users) version=4 mtime=2018-02-15 20:45:51.489653481 +0000 UTC
I180215 20:45:51.630626 25912 storage/replica_command.go:838  [split,n1,s1,r19/1:/{Table/22-Max}] initiating a split of this range at key /Table/23 [r20]
W180215 20:45:51.786895 25916 storage/intent_resolver.go:320  [n1,s1] failed to push during intent resolution: failed to push "split" id=164ad23f key=/Local/Range/Table/22/RangeDescriptor rw=true pri=0.00522989 iso=SERIALIZABLE stat=PENDING epo=0 ts=1518727551.630845456,0 orig=1518727551.630845456,0 max=1518727551.630845456,0 wto=false rop=false seq=3
I180215 20:45:51.822392 25430 sql/event_log.go:114  [n1] Event: "finish_schema_change", target: 4, info: {MutationID:1}
I180215 20:45:51.856140 25430 sql/lease.go:275  publish (count leases): descID=4 name=users version=3 count=1
I180215 20:45:51.887307 25989 sql/lease.go:800  new lease: 4("users") ver=4:1518727806.434029774,0, refcount=0
I180215 20:45:51.931253 25430 sql/lease.go:275  publish (count leases): descID=4 name=users version=3 count=1
I180215 20:45:52.523151 25430 server/server.go:1254  [n1] done ensuring all necessary migrations have run
I180215 20:45:52.523346 25430 server/server.go:1257  [n1] serving sql connections
I180215 20:45:52.628929 25952 sql/event_log.go:114  [n1] Event: "node_join", target: 1, info: {Descriptor:{NodeID:1 Address:{NetworkField:tcp AddressField:127.0.0.1:33565} Attrs: Locality: ServerVersion:1.1-12} ClusterID:73107500-011e-490e-b0c6-053a9b2d599b StartedAt:1518727547839393362 LastUp:1518727547839393362}
I180215 20:45:52.993902 25964 storage/replica_command.go:838  [split,n1,s1,r20/1:/{Table/23-Max}] initiating a split of this range at key /Table/50 [r21]
I180215 20:45:53.029558 25962 sql/event_log.go:114  [client=127.0.0.1:60458,user=root,n1] Event: "create_database", target: 50, info: {DatabaseName:test Statement:CREATE DATABASE IF NOT EXISTS test User:root}
I180215 20:45:53.224882 25965 storage/replica_command.go:838  [split,n1,s1,r21/1:/{Table/50-Max}] initiating a split of this range at key /Table/51 [r22]
I180215 20:45:53.243403 25962 sql/event_log.go:114  [client=127.0.0.1:60458,user=root,n1] Event: "create_table", target: 51, info: {TableName:test.scan Statement:CREATE TABLE test.scan (a INT, b INT, v STRING, PRIMARY KEY (a, b)) User:root}
W180215 20:45:53.380561 26050 storage/intent_resolver.go:320  [n1,s1] failed to push during intent resolution: failed to push "split" id=1b04c15a key=/Local/Range/Table/50/RangeDescriptor rw=true pri=0.01770058 iso=SERIALIZABLE stat=PENDING epo=0 ts=1518727553.225189310,0 orig=1518727553.225189310,0 max=1518727553.225189310,0 wto=false rop=false seq=3
I180215 20:45:56.308420 25590 storage/replica_proposal.go:196  [n1,s1,r22/1:/{Table/51-Max}] new range lease repl=(n1,s1):1 seq=2 start=0.000000000,0 epo=1 pro=1518727556.304102206,0 following repl=(n1,s1):1 seq=1 start=0.000000000,0 exp=1518727556.800523088,0 pro=1518727547.800616679,0
I180215 20:45:56.397743 25572 storage/replica_proposal.go:196  [n1,s1,r12/1:/Table/1{5-6}] new range lease repl=(n1,s1):1 seq=2 start=0.000000000,0 epo=1 pro=1518727556.393261485,0 following repl=(n1,s1):1 seq=1 start=0.000000000,0 exp=1518727556.800523088,0 pro=1518727547.800616679,0
I180215 20:45:56.599722 25585 storage/replica_proposal.go:196  [n1,s1,r18/1:/Table/2{1-2}] new range lease repl=(n1,s1):1 seq=2 start=0.000000000,0 epo=1 pro=1518727556.595558958,0 following repl=(n1,s1):1 seq=1 start=0.000000000,0 exp=1518727556.800523088,0 pro=1518727547.800616679,0
I180215 20:45:56.810332 25604 storage/replica_proposal.go:196  [n1,s1,r17/1:/Table/2{0-1}] new range lease repl=(n1,s1):1 seq=2 start=0.000000000,0 epo=1 pro=1518727556.802097983,0 following repl=(n1,s1):1 seq=1 start=0.000000000,0 exp=1518727556.800523088,0 pro=1518727547.800616679,0
I180215 20:45:57.065767 25593 storage/replica_proposal.go:196  [n1,s1,r7/1:/Table/{SystemCon…-11}] new range lease repl=(n1,s1):1 seq=2 start=0.000000000,0 epo=1 pro=1518727557.061344807,0 following repl=(n1,s1):1 seq=1 start=0.000000000,0 exp=1518727556.800523088,0 pro=1518727547.800616679,0
==================
Write at 0x000003dee838 by goroutine 862:
  github.com/cockroachdb/cockroach/pkg/sql/sqlbase.SetKVBatchSize()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/sqlbase/kvfetcher.go:138 +0x5d
  github.com/cockroachdb/cockroach/pkg/sql.TestScanBatches()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/scan_test.go:178 +0xbeb
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:746 +0x16c

Previous read at 0x000003dee838 by goroutine 550:
  github.com/cockroachdb/cockroach/pkg/sql/sqlbase.(*txnKVFetcher).fetch()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/sqlbase/kvfetcher.go:268 +0x1324
  github.com/cockroachdb/cockroach/pkg/sql/sqlbase.(*txnKVFetcher).nextBatch()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/sqlbase/kvfetcher.go:359 +0xbb
  github.com/cockroachdb/cockroach/pkg/sql/sqlbase.(*RowFetcher).nextKV()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/sqlbase/rowfetcher.go:419 +0x22e
  github.com/cockroachdb/cockroach/pkg/sql/sqlbase.(*RowFetcher).NextKey()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/sqlbase/rowfetcher.go:435 +0x98
  github.com/cockroachdb/cockroach/pkg/sql/sqlbase.(*RowFetcher).StartScanFrom()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/sqlbase/rowfetcher.go:409 +0x111
  github.com/cockroachdb/cockroach/pkg/sql/sqlbase.(*RowFetcher).StartScan()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/sqlbase/rowfetcher.go:399 +0x29a
  github.com/cockroachdb/cockroach/pkg/sql.(*scanNode).initScan()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/scan.go:222 +0x203
  github.com/cockroachdb/cockroach/pkg/sql.(*scanNode).Next()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/scan.go:182 +0x486
  github.com/cockroachdb/cockroach/pkg/sql.(*renderNode).Next()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/render.go:338 +0xad
  github.com/cockroachdb/cockroach/pkg/sql.(*groupNode).Next()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/group.go:362 +0xead
  github.com/cockroachdb/cockroach/pkg/sql.forEachRow()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/executor.go:1978 +0x79
  github.com/cockroachdb/cockroach/pkg/sql.(*planner).queryRows()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/planner.go:380 +0x497
  github.com/cockroachdb/cockroach/pkg/sql.(*planner).QueryRow()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/planner.go:338 +0xab
  github.com/cockroachdb/cockroach/pkg/sql.LeaseStore.countLeases.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/lease.go:405 +0x36c
  github.com/cockroachdb/cockroach/pkg/internal/client.(*DB).Txn.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/db.go:515 +0x5e
  github.com/cockroachdb/cockroach/pkg/internal/client.(*Txn).Exec()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/txn.go:738 +0x128
  github.com/cockroachdb/cockroach/pkg/internal/client.(*DB).Txn()
      /go/src/github.com/cockroachdb/cockroach/pkg/internal/client/db.go:514 +0x183
  github.com/cockroachdb/cockroach/pkg/sql.LeaseStore.countLeases()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/lease.go:399 +0x1f0
  github.com/cockroachdb/cockroach/pkg/sql.LeaseStore.WaitForOneVersion()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/lease.go:268 +0x74c
  github.com/cockroachdb/cockroach/pkg/sql.LeaseStore.Publish()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/lease.go:304 +0x224
  github.com/cockroachdb/cockroach/pkg/sql.(*SchemaChanger).MaybeIncrementVersion()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/schema_changer.go:509 +0x122
  github.com/cockroachdb/cockroach/pkg/sql.(*SchemaChanger).exec()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/schema_changer.go:396 +0x1e4
  github.com/cockroachdb/cockroach/pkg/sql.(*SchemaChangeManager).Start.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/schema_changer.go:1084 +0x9d6
  github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:195 +0x160

Goroutine 862 (running) created at:
  testing.(*T).Run()
      /usr/local/go/src/testing/testing.go:789 +0x568
  testing.runTests.func1()
      /usr/local/go/src/testing/testing.go:1004 +0xa7
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:746 +0x16c
  testing.runTests()
      /usr/local/go/src/testing/testing.go:1002 +0x521
  testing.(*M).Run()
      /usr/local/go/src/testing/testing.go:921 +0x206
  github.com/cockroachdb/cockroach/pkg/sql_test.TestMain()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/main_test.go:36 +0x15d
  main.main()
      github.com/cockroachdb/cockroach/pkg/sql/_test/_testmain.go:480 +0x1cd

Goroutine 550 (running) created at:
  github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:188 +0xba
  github.com/cockroachdb/cockroach/pkg/sql.(*SchemaChangeManager).Start()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/schema_changer.go:979 +0x12f
  github.com/cockroachdb/cockroach/pkg/server.(*Server).Start()
      /go/src/github.com/cockroachdb/cockroach/pkg/server/server.go:1165 +0x43e0
  github.com/cockroachdb/cockroach/pkg/server.(*TestServer).Start()
      /go/src/github.com/cockroachdb/cockroach/pkg/server/testserver.go:336 +0x24c
  github.com/cockroachdb/cockroach/pkg/testutils/serverutils.StartServerRaw()
      /go/src/github.com/cockroachdb/cockroach/pkg/testutils/serverutils/test_server_shim.go:204 +0x143
  github.com/cockroachdb/cockroach/pkg/testutils/serverutils.StartServer()
      /go/src/github.com/cockroachdb/cockroach/pkg/testutils/serverutils/test_server_shim.go:172 +0x73
  github.com/cockroachdb/cockroach/pkg/sql.TestScanBatches()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/scan_test.go:131 +0xc5
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:746 +0x16c
==================
?   	github.com/cockroachdb/cockroach/pkg/sql/coltypes	[no test files]
I180215 20:40:52.815320 1 rand.go:74  Random seed: 3295516578724522379
------- Stdout: -------
W180215 20:38:31.115794 33808 server/status/runtime.go:109  Could not parse build timestamp: parsing time "" as "2006/01/02 15:04:05": cannot parse "" as "2006"
I180215 20:38:31.123519 33808 server/config.go:522  [n?] 1 storage engine initialized
I180215 20:38:31.123551 33808 server/config.go:525  [n?] RocksDB cache size: 128 MiB
I180215 20:38:31.123559 33808 server/config.go:525  [n?] store 0: in-memory, size 0 B
I180215 20:38:31.126688 33808 server/node.go:374  [n?] **** cluster 34869c82-4d55-49f9-9cc1-3c2531d7949a has been created
I180215 20:38:31.126732 33808 server/server.go:979  [n?] **** add additional nodes by specifying --join=127.0.0.1:41187
I180215 20:38:31.127765 33808 storage/store.go:1307  [n1,s1] [n1,s1]: failed initial metrics computation: [n1,s1]: system config not yet available
I180215 20:38:31.127833 33808 server/node.go:504  [n1] initialized store [n1,s1]: disk (capacity=512 MiB, available=512 MiB, used=0 B, logicalBytes=4.9 KiB), ranges=1, leases=0, writes=0.00, bytesPerReplica={p10=5045.00 p25=5045.00 p50=5045.00 p75=5045.00 p90=5045.00}, writesPerReplica={p10=0.00 p25=0.00 p50=0.00 p75=0.00 p90=0.00}
I180215 20:38:31.127862 33808 server/node.go:352  [n1] node ID 1 initialized
I180215 20:38:31.127925 33808 gossip/gossip.go:332  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:41187" > attrs:<> locality:<> ServerVersion:<major_val:1 minor_val:1 patch:0 unstable:12 > 
I180215 20:38:31.128161 33808 storage/stores.go:331  [n1] read 0 node addresses from persistent storage
I180215 20:38:31.128239 33808 server/node.go:645  [n1] connecting to gossip network to verify cluster ID...
I180215 20:38:31.130179 33808 server/node.go:670  [n1] node connected via gossip and verified as part of cluster "34869c82-4d55-49f9-9cc1-3c2531d7949a"
I180215 20:38:31.130208 33808 server/node.go:446  [n1] node=1: started with [<no-attributes>=<in-mem>] engine(s) and attributes []
I180215 20:38:31.130302 33808 server/server.go:1180  [n1] starting https server at 127.0.0.1:41539
I180215 20:38:31.130315 33808 server/server.go:1181  [n1] starting grpc/postgres server at 127.0.0.1:41187
I180215 20:38:31.130327 33808 server/server.go:1182  [n1] advertising CockroachDB node at 127.0.0.1:41187
I180215 20:38:31.132871 34290 storage/replica_command.go:838  [split,n1,s1,r1/1:/M{in-ax}] initiating a split of this range at key /System/"" [r2]
I180215 20:38:31.159637 34106 storage/replica_command.go:838  [split,n1,s1,r2/1:/{System/-Max}] initiating a split of this range at key /System/NodeLiveness [r3]
W180215 20:38:31.160284 34257 storage/intent_resolver.go:320  [n1,s1] failed to push during intent resolution: failed to push "sql txn implicit" id=039b0fb3 key=/Table/SystemConfigSpan/Start rw=true pri=0.02516969 iso=SERIALIZABLE stat=PENDING epo=0 ts=1518727111.132184592,0 orig=1518727111.132184592,0 max=1518727111.132184592,0 wto=false rop=false seq=7
I180215 20:38:31.161241 33808 sql/event_log.go:114  [n1] Event: "set_cluster_setting", target: 0, info: {SettingName:diagnostics.reporting.enabled Value:true User:node}
I180215 20:38:31.170534 33808 sql/event_log.go:114  [n1] Event: "set_cluster_setting", target: 0, info: {SettingName:version Value:$1 User:node}
I180215 20:38:31.171813 34310 storage/replica_command.go:838  [split,n1,s1,r3/1:/{System/NodeL…-Max}] initiating a split of this range at key /System/NodeLivenessMax [r4]
I180215 20:38:31.176375 34333 storage/replica_command.go:838  [split,n1,s1,r4/1:/{System/NodeL…-Max}] initiating a split of this range at key /System/tsd [r5]
W180215 20:38:31.178113 34331 storage/intent_resolver.go:320  [n1,s1] failed to push during intent resolution: failed to push "sql txn implicit" id=0541193b key=/Table/SystemConfigSpan/Start rw=true pri=0.03092177 iso=SERIALIZABLE stat=PENDING epo=0 ts=1518727111.172675487,0 orig=1518727111.172675487,0 max=1518727111.172675487,0 wto=false rop=false seq=7
I180215 20:38:31.178838 33808 sql/event_log.go:114  [n1] Event: "set_cluster_setting", target: 0, info: {SettingName:trace.debug.enable Value:false User:node}
I180215 20:38:31.182205 33928 storage/replica_command.go:838  [split,n1,s1,r5/1:/{System/tsd-Max}] initiating a split of this range at key /System/"tse" [r6]
I180215 20:38:31.188245 31889 storage/replica_command.go:838  [split,n1,s1,r6/1:/{System/tse-Max}] initiating a split of this range at key /Table/SystemConfigSpan/Start [r7]
I180215 20:38:31.193505 34313 storage/replica_command.go:838  [split,n1,s1,r7/1:/{Table/System…-Max}] initiating a split of this range at key /Table/11 [r8]
I180215 20:38:31.207577 34348 storage/replica_command.go:838  [split,n1,s1,r8/1:/{Table/11-Max}] initiating a split of this range at key /Table/12 [r9]
I180215 20:38:31.223119 34392 storage/replica_command.go:838  [split,n1,s1,r9/1:/{Table/12-Max}] initiating a split of this range at key /Table/13 [r10]
I180215 20:38:31.232392 33808 sql/event_log.go:114  [n1] Event: "alter_table", target: 4, info: {TableName:users Statement:ALTER TABLE system.users ADD COLUMN IF NOT EXISTS "isRole" BOOL NOT NULL DEFAULT false User:node MutationID:1 CascadeDroppedViews:[]}
I180215 20:38:31.233849 34418 storage/replica_command.go:838  [split,n1,s1,r10/1:/{Table/13-Max}] initiating a split of this range at key /Table/14 [r11]
I180215 20:38:31.235502 33808 sql/lease.go:349  [n1] publish: descID=4 (users) version=2 mtime=2018-02-15 20:38:31.235372468 +0000 UTC
I180215 20:38:31.239610 34436 storage/replica_command.go:838  [split,n1,s1,r11/1:/{Table/14-Max}] initiating a split of this range at key /Table/15 [r12]
I180215 20:38:31.245758 33808 sql/lease.go:349  [n1] publish: descID=4 (users) version=3 mtime=2018-02-15 20:38:31.245446789 +0000 UTC
I180215 20:38:31.247889 34430 storage/replica_command.go:838  [split,n1,s1,r12/1:/{Table/15-Max}] initiating a split of this range at key /Table/16 [r13]
I180215 20:38:31.254018 34401 storage/replica_command.go:838  [split,n1,s1,r13/1:/{Table/16-Max}] initiating a split of this range at key /Table/17 [r14]
I180215 20:38:31.254734 33808 sql/backfill.go:134  [n1] Running backfill for "users", v=3, m=1
I180215 20:38:31.260125 34454 storage/replica_command.go:838  [split,n1,s1,r14/1:/{Table/17-Max}] initiating a split of this range at key /Table/18 [r15]
I180215 20:38:31.265135 34409 storage/replica_command.go:838  [split,n1,s1,r15/1:/{Table/18-Max}] initiating a split of this range at key /Table/19 [r16]
I180215 20:38:31.269801 34530 storage/replica_command.go:838  [split,n1,s1,r16/1:/{Table/19-Max}] initiating a split of this range at key /Table/20 [r17]
I180215 20:38:31.275616 34543 storage/replica_command.go:838  [split,n1,s1,r17/1:/{Table/20-Max}] initiating a split of this range at key /Table/21 [r18]
I180215 20:38:31.288069 34461 storage/replica_command.go:838  [split,n1,s1,r18/1:/{Table/21-Max}] initiating a split of this range at key /Table/22 [r19]
I180215 20:38:31.288719 33808 sql/lease.go:349  [n1] publish: descID=4 (users) version=4 mtime=2018-02-15 20:38:31.288627233 +0000 UTC
I180215 20:38:31.292755 33808 sql/event_log.go:114  [n1] Event: "finish_schema_change", target: 4, info: {MutationID:1}
I180215 20:38:31.295584 33808 sql/lease.go:275  publish (count leases): descID=4 name=users version=3 count=1
I180215 20:38:31.295670 34563 sql/lease.go:800  new lease: 4("users") ver=4:1518727467.883483686,0, refcount=0
I180215 20:38:31.297620 34557 storage/replica_command.go:838  [split,n1,s1,r19/1:/{Table/22-Max}] initiating a split of this range at key /Table/23 [r20]
I180215 20:38:31.349514 33808 server/server.go:1254  [n1] done ensuring all necessary migrations have run
I180215 20:38:31.349551 33808 server/server.go:1257  [n1] serving sql connections
I180215 20:38:31.357351 34372 sql/event_log.go:114  [n1] Event: "node_join", target: 1, info: {Descriptor:{NodeID:1 Address:{NetworkField:tcp AddressField:127.0.0.1:41187} Attrs: Locality: ServerVersion:1.1-12} ClusterID:34869c82-4d55-49f9-9cc1-3c2531d7949a StartedAt:1518727111130187789 LastUp:1518727111130187789}
I180215 20:38:31.377502 34490 sql/event_log.go:114  [client=127.0.0.1:40316,user=root,n1] Event: "create_database", target: 50, info: {DatabaseName:test Statement:CREATE DATABASE IF NOT EXISTS test User:root}
I180215 20:38:31.377936 34510 storage/replica_command.go:838  [split,n1,s1,r20/1:/{Table/23-Max}] initiating a split of this range at key /Table/50 [r21]
I180215 20:38:31.381410 34490 sql/event_log.go:114  [client=127.0.0.1:40316,user=root,n1] Event: "create_table", target: 51, info: {TableName:test.scan Statement:CREATE TABLE test.scan (a INT, b INT, v STRING, PRIMARY KEY (a, b)) User:root}
I180215 20:38:31.382213 34588 storage/replica_command.go:838  [split,n1,s1,r21/1:/{Table/50-Max}] initiating a split of this range at key /Table/51 [r22]
I180215 20:38:31.526522 34490 sql/event_log.go:114  [client=127.0.0.1:40316,user=root,n1] Event: "drop_table", target: 51, info: {TableName:scan Statement:DROP TABLE test.scan User:root CascadeDroppedViews:[]}
I180215 20:38:31.543700 34490 sql/lease.go:349  [client=127.0.0.1:40316,user=root,n1] publish: descID=51 (scan) version=2 mtime=2018-02-15 20:38:31.543536251 +0000 UTC
I180215 20:38:31.545328 34490 sql/lease.go:349  [client=127.0.0.1:40316,user=root,n1] publish: descID=51 (scan) version=3 mtime=2018-02-15 20:38:31.545215374 +0000 UTC

```

Please assign, take a look and update the issue accordingly.

--- a/pkg/sql/scan_test.go
+++ b/pkg/sql/scan_test.go
@@ -128,6 +128,10 @@ func testScanBatchQuery(t *testing.T, db *gosql.DB, numSpans, numAs, numBs int, 
 func TestScanBatches(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 
+	// The test will screw around with KVBatchSize; make sure to restore it at the end.
+	restore := sqlbase.SetKVBatchSize(10)
+	defer restore()
+
 	s, db, _ := serverutils.StartServer(
 		t, base.TestServerArgs{UseDatabase: "test"})
 	defer s.Stopper().Stop(context.TODO())
@@ -135,10 +139,6 @@ func TestScanBatches(t *testing.T) {
 	if _, err := db.Exec(`CREATE DATABASE IF NOT EXISTS test`); err != nil {
 		t.Fatal(err)
 	}
-
-	// The test will screw around with KVBatchSize; make sure to restore it at the end.
-	restore := sqlbase.SetKVBatchSize(10)
-	defer restore()
 
 	numAs := 5
 	numBs := 20

